src/components/MovieList/index.js

```diff
@@ -1,47 +1,38 @@
 import React from 'react';
-import { StyleSheet, Text, FlatList, Image, View } from 'react-native';
-import { Config } from 'react-native-config';
+import { StyleSheet, Text, FlatList, View, TouchableOpacity } from 'react-native';
 import { useTheme } from '@react-navigation/native';
 import { typography, spacing } from '@/theme';
-import { movie } from '@/assets';
+import { MovieItem, MessageBanner } from '@/components';
+import { NAVIGATION } from '@/constants';
 
-export const MovieList = ({ item }) => {
+export const MovieList = ({ navigation, item }) => {
   const { colors } = useTheme();
 
-  const renderCarousel = (movieList) => {
-    return (
-      <>
-        {movieList.item.poster_path ? (
-          <Image
-            source={{ uri: Config.IMAGE_API_BASE_URL + movieList.item.poster_path }}
-            resizeMode="cover"
-            style={styles.img}
-            accessibilityIgnoresInvertColors
-          />
-        ) : (
-          <View
-            style={[
-              styles.img,
-              styles.unknownContainer,
-              { backgroundColor: colors.posterBackground },
-            ]}
-          >
-            <Image
-              source={movie}
-              resizeMode="cover"
-              style={styles.unknownImg}
-              accessibilityIgnoresInvertColors
-            />
-          </View>
-        )}
-      </>
-    );
+  const handlePress = () => {
+    navigation.navigate(NAVIGATION.movieCollection, {
+      collection: item,
+      noObjectMessage: "There's no movie added to My List yet",
+    });
   };
 
   return (
     <View style={styles.container}>
-      <Text style={[styles.title, { color: colors.text }, typography.title]}>{item.title}</Text>
-      <FlatList data={item.movieList} renderItem={renderCarousel} horizontal />
+      <TouchableOpacity onPress={handlePress}>
+        <Text style={[styles.title, { color: colors.text }, typography.title]}>{item.title}</Text>
+      </TouchableOpacity>
+      {item.movieList.length ? (
+        <FlatList
+          data={item.movieList}
+          renderItem={(movieItem) => <MovieItem item={movieItem} customStyles={movieItemStyles} />}
+          horizontal
+        />
+      ) : (
+        <MessageBanner
+          customStyles={{
+            messageContainer: { ...styles.messageContainer, backgroundColor: colors.card },
+          }}
+        />
```

Comment:
This is okay. Yust for you to know (and if you want to change it even better), checkout ListEmptyComponent flatlist option